src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -0,0 +1,177 @@
+        self.msConfig.setdefault("mockMongoDB", False)
+
+        # A full set of valid database connection parameters can be found at:
+        # https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html
+        mongoIndex = IndexModel('pileupName', unique=True)
+        collection = self.msConfig.get('mongoDBCollection', 'msPileupCollection')
+        msOutDBConfig = {
+            'database': self.msConfig['mongoDB'],
+            'server': self.msConfig['mongoDBServer'],
+            'replicaSet': self.msConfig['mongoDBReplicaSet'],
+            'port': self.msConfig['mongoDBPort'],
+            'username': self.msConfig['mongoDBUser'],
+            'password': self.msConfig['mongoDBPassword'],
+            'mockMongoDB': self.msConfig['mockMongoDB'],
+            'connect': True,
+            'directConnection': False,
+            'logger': self.logger,
+            'create': True,
+            'collections': [(collection, mongoIndex)]
+        }
+        mongoDB = MongoDB(**msOutDBConfig)
+        self.msDB = getattr(mongoDB, self.msConfig['mongoDB'])
+        self.dbColl = self.msDB[collection]
+#         self.dbColl.create_indexes([mongoIndex])
+
+    def create(self, cdict):
+        """
+        Create and return pileup data from campaigns dictionary
+
+        :param cdict: a dictionary of campaigns from CMSCompOps/WmAgentScript/campaign.json
+        :return: a pileup data disctionary
+        """
+        return
+
+    def update(self, doc):
+        """
+        Update pileup data with provided input
+
+        :param doc: a dictionary of pieleup data to be updated
+        :return: None
+        """
+        try:
+            self.dbColl.upsert(doc)
+        except Exception as exp:
+            msg = f"Failed to insert: {doc}, error {exp}"
+            self.logger.exception(msg)
+    
+    def store(self, doc):
+        """
+        Store data to persistent storage, e.g. database
+
+        :param doc: input data
+        :return: None
+        """
+        try:
+            self.dbColl.insert_one(doc)
+        except errors.DuplicateKeyError:
+            msg = f"Failed to insert: {doc}, it already exist in DB"
+            self.logger.exception(msg)
+
+    def get(self, pileupId):
+        """
+        fetches MSPileup data from persistent storage for a given pileupID
+
+        :param pileupId: input pileup ID (integer)
+        :return: list of result documents fetched from underlying persisten storage (MongoDB)
+        """
+        spec = {'pileupId': pileupId}
+        result = []
+
+        doc = self.dbColl.find_one(spec)
+        result.append(stripKeys(doc))
+        return result
+
+    def delete(self, pileupId):
+        """
+        deletes MSPileup data in persistent storage for given pileupID
+
+        :param pileupId: input pileup ID (integer)
+        :return: None
+        """
+        spec = {'pileupId': pileupId}
+        self.dbColl.remove(spec)
+
+    def validate(self, pdict):
```

Comment:
this document validation also may get a similar approach to the way how we define those data structures in other micro services. Meaning if we have  a doc schema defined in the `_init_`  we could simply iterate  through it to check data types. similar to: 
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleanerWflow.py#L134-L230